pandas/io/pytables.py

```diff
@@ -4842,6 +4774,74 @@ def _unconvert_index(data, kind: str, encoding=None, errors="strict"):
     return index
 
 
+def _maybe_convert_for_string_atom(
+    name: str, block, existing_col, min_itemsize, nan_rep, encoding, errors
```

Comment:
just cut / paste?

add a doc-string / types at some point

Reply:
Mostly cut/paste, but from a couple different places instead of one big chunk.  The `not block.is_object` check on 4782 is new, wasnt needed before because before we were doing things in a different order.
The only real change is the `reshape` on 4827 and assertion on 4828 are new.